geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala

```diff
@@ -161,12 +155,12 @@ object SelfClosingIterator {
         iterHasNext
       }
       override def next(): A = iter.next()
-      override def close(): Unit = closeIter()
+      override def close(): Unit = closeIter
     }
 
-  def apply[A](iter: Iterator[A] with Closeable): SelfClosingIterator[A] = apply(iter, iter.close)
+  def apply[A](iter: Iterator[A] with Closeable): SelfClosingIterator[A] = apply(iter, iter.close())
```

Comment:
Ah, ok, thanks for the unit test.  I missed the both old and new versions are 'productions' with =>.  I feel better now:).